### DISCUSSION TITLE
support configurable start port instead of a magic number 9000

### DISCUSSION THREAD

**Discussion by ladyrick** [microsoft/vscode-jupyter#10761] [New Ideas]
`KernelLauncher.computeStartPort` support different start port instead of a magic number `9000`:
https://github.dev/microsoft/vscode-jupyter/blob/be3667521cf217753b343c7483c128b34dd5d213/src/kernels/raw/launcher/kernelLauncher.node.ts#L99

In my case, I need port 9000-10000 to be reserved. These ports may be not used now but may be used in the future. Now jupyter uses 9000 as start port. It's very likely to occupy a port in 9000-10000 and later my other application would failed because of "port is already in use".

**Comment by amunger** [comment]
sounds like a reasonable request, I created feature request #10765 from this discussion to track it
